### PR TITLE
[RFC] clang-format

### PR DESCRIPTION
**Pull request by neheb**

### CODE REVIEW

**Comment by ktmf01** [comment]
@neheb I guess this PR is no longer applicable? Travis is no longer in use here. Should this PR be closed?

**Comment by neheb** [comment]
the first commit is somewhat valid. The second is completely useless. I'll rebase.

File: .clang-format [hunk 1]
```diff
@@ -0,0 +1,19 @@
+BasedOnStyle: WebKit
+AlignAfterOpenBracket: Align
+AlignEscapedNewlines: DontAlign
+AlignOperands: true
+BraceWrapping:
+  AfterFunction: true
```
Comment (ktmf01) [inline .clang-format:6]:
```suggestion
  AfterFunction: true
  BeforeElse: true

```

File: .clang-format [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+BreakBeforeBinaryOperators: None
+BreakBeforeBraces: Custom
+BreakBeforeInheritanceComma: true
+BreakConstructorInitializers: BeforeComma
+DerivePointerAlignment: false
```
Comment (ktmf01) [inline .clang-format:11]:
```suggestion
DerivePointerAlignment: false
IndentCaseLabels: true
```

**Review by ktmf01 — COMMENTED** [review]
Here are two additions.

Still, this doesn't really fit with FLACs coding style. This is apparently a problem with clang-format, see [here](https://stackoverflow.com/questions/67245250/can-i-tell-clang-format-to-not-add-space-around-arithmetic-operators) and [here](https://stackoverflow.com/questions/44259984/how-to-tell-clang-format-to-keep-whitespace-between-binary-operators-as-they-are).

Also, when formatting lpc.c, clang-format IMO makes a lot of fall-through switch-case less easy to read.

**Comment by neheb** [comment]
Added changes. git diff on stream_decoder.c goes down by half. cool.

**Comment by ktmf01** [comment]
Are you planning on improving this even further, or can this be merged? It is still marked 'draft'.

I agree with erikd in #208 though, I don't think it is a good idea to clang-format the whole tree. However, this file is nice to have to use with git-clang-format.

**Comment by neheb** [comment]
Marked as ready.

**Pull request merged**